src/Mod/Arch/ArchWall.py

```diff
@@ -1205,7 +1205,10 @@ def getExtrusionData(self,obj):
         if not height:
             return None
         if obj.Normal == Vector(0,0,0):
-            normal = Vector(0,0,1)
+            import DraftGeomUtils
+            normal = DraftGeomUtils.get_shape_normal(obj.Base.Shape)
```

Comment:
I fear this might break a number of existing models.. Might need some thorough testing

Reply:
Hmmm, the original code would not work e.g. model in FC discussion reported bug :
- https://forum.freecad.org/viewtopic.php?t=81989
and return 
-  DraftGeomUtils: unable to bind wires
-  Wall001 has a null shape

In general, wires not parallel to XY plane would not work.

@Roy-043 Indeed there is 1 case that there are 2 normal direction would work - a wire / sketch with 1 edge not parallel to XY plane, then below normal works :
-  both (0,0,1) 
-  the normal the currently proposed code deduced : the one perpendicular to the edge but align with Global Z-direction

Any comment? :)  Do not expect case like that though?

Reply:
A single edge is of course problematic. `get_shape_normal` should then derive the normal from the container of the edge. Using its LCS (local coordinate system).

If you use the GCS (global CS) instead, the shape built on the edge would change if the orientation of the object changes.

IMO we should get the wall tool to work on any plane.